lib/commands/init.js

```diff
@@ -107,17 +107,17 @@ module.exports = Command.extend({
         }
       })
       .then(function() {
-        if (!commandOptions.skipNpm) {
-          return npmInstall.run({
-            verbose: commandOptions.verbose,
-            optional: false
+        if (!commandOptions.skipBower) {
+          return bowerInstall.run({
+            verbose: commandOptions.verbose
```

Comment:
having the generation command as a postinstall hook means it gets executed when you create a new ember-cli project and npmInstall.run() executes. Which is great but any ember command will fail if bowerInstall did not run before. So here i make sure that first all bower deps are installed, then all node modules and only **then** `ember update-package-cache` gets called